src/Utilities.jl

```diff
@@ -0,0 +1,27 @@
+###-----------------------------------------------------------------------------
+### Copyright (C) 2022- Cursor Insight
+###
+### All rights reserved.
+###-----------------------------------------------------------------------------
+
+module Utilities
+
+###=============================================================================
+### Functions
+###=============================================================================
+
+to_str(::Type{Val{x}}) where {x} = ":$x"
+to_str(::Type{T}) where {T} = "::$T"
+
+symbol_to_val(s::Symbol) = Val(s)
+symbol_to_val(x) = x
+
+default(f) = Tuple{typeof(f), Vararg{Any}}
+
+function patterns(f)::Vector
+    return filter(getproperty.(methods(f).ms, :sig)) do signature::Type
+        return !(default(f) <: signature)
+    end
```

Comment:
Nitpicking.

In my experience, mixing broadcasting, `filter` and (implicit) `map` makes the code difficult to understand, and it is better to transform it into a single list comprehension, such as:

```jl
return [method.sig for method in methods(f).ms
        if !(default(f) <: method.sig)]
```